polaris-react/src/components/Text/Text.tsx

```diff
@@ -58,6 +60,8 @@ export interface TextProps {
   variant?: Variant;
   /** Visually hide the text */
   visuallyHidden?: boolean;
+  /** Add a underline or line-through style to the text */
+  decoration?: Decoration;
```

Comment:
I can change the prop name if that makes more sense. 
I'm not sure if we would want to make this an exact copy of the text decoration option, I only added the 2 options based on what I found to be common in WEB, but I don't know if we would want to allow much more than that, sounds like a Polaris answer to me?

Reply:
I've seen a couple use cases of `text-decoration-style: dotted` in web. It may make sense to split this out into `textDecorationLine` and `textDecorationStyle` props to reduce the need to wrap `Text` components with a `span` to apply `dotted` styles.

Reply:
Implementing both now, I wonder from a design perspective, what options should I include?

```
/* Keyword values */
text-decoration-style: solid;
text-decoration-style: double;
text-decoration-style: dotted;
text-decoration-style: dashed;
text-decoration-style: wavy;

/* Global values */
text-decoration-style: inherit;
text-decoration-style: initial;
text-decoration-style: revert;
text-decoration-style: revert-layer;
text-decoration-style: unset;
```

```
/* Single keyword */
text-decoration-line: none;
text-decoration-line: underline;
text-decoration-line: overline;
text-decoration-line: line-through;
text-decoration-line: blink;

/* Multiple keywords */
text-decoration-line: underline overline; /* Two decoration lines */
text-decoration-line: overline underline line-through; /* Multiple decoration lines */

/* Global values */
text-decoration-line: inherit;
text-decoration-line: initial;
text-decoration-line: revert;
text-decoration-line: revert-layer;
text-decoration-line: unset;
```

How far down the rabbit whole should we go?

Reply:
I think we can limit it to the `text-decoration-line: underline | line-through` and `text-decoration-style: solid | dotted`. I haven't seen other styles used throughout web and if a use case comes up for it globally, the component can be supplemented in a later PR.

Reply:
Thanks, makes sense. I'll apply these changes and re-request review :)